src/commands/run.js

```diff
@@ -0,0 +1,89 @@
+const log = require('@dhis2/cli-helpers-engine').reporter
+const spawn = require('cross-spawn')
+const fs = require('fs-extra')
+const path = require('path')
+
+module.exports.command = 'run <package> <codemod> <files..>'
+module.exports.alias = 'r'
+module.exports.desc = `* Runs a jscodemod.
+* Packages must be in the "node_modules" dir
+* Codemods must be located in "codemods" folders
+
+Example:
+* run @dhis2/cli-utils-cypress name-of-a-codemod cypress/integration/**/*.js --forwardArgs key1=val1 key2=val2
+-> Will run the "name-of-a-codemod.js"
+-> Located in ./node_modules/@dhis2/cli-utils-cypress/codemods/name-of-a-codemod.js
+-> Will apply codemod to all js files located in cypress/integration
+-> Will forward { key1: 'val1', key2: 'val2' } to the codemod
+`
+
+module.exports.builder = yargs =>
+    yargs
+        .positional('package', {
+            describe: 'The name of the package that contains the codemod',
+            type: 'string',
+        })
+        .positional('codemod', {
+            describe: 'Name of the codemod file',
+            type: 'string',
+        })
+        .positional('files', {
+            describe: 'Files or path the codemod should be applied to',
+            type: 'string',
+        })
+        .option('package-folder', {
+            describe: 'The package the folder can be found in',
+            type: 'string',
+            default: path.join(`${process.cwd()}`, 'node_modules')
+        })
+        .option('codemods-folder', {
+            describe: 'The package the codemods can be found in',
+            type: 'string',
+            default: 'codemods',
+        })
+        .option('forward-args', {
+            describe: 'Args that should be forwarded to the codemod',
+            type: 'array',
+            default: [],
+        })
+
+module.exports.handler = argv => {
+    const {
+        codemod,
+        codemodsFolder,
+        files,
+        forwardArgs,
+        package,
+        packageFolder,
+    } = argv
+
+    const packagePath = path.join(packageFolder, package)
+    if (!fs.pathExistsSync(packagePath)) {
+        log.error(`The package "${package}" is not installed. Check your package.json if it's included.`)
+        process.exit(1)
+    }
+
+    const codemodsPath = path.join(packagePath, codemodsFolder)
+    if (!fs.pathExistsSync(codemodsPath)) {
+        log.error(`The package "${package}" does not contain any codemods.`)
+        process.exit(1)
+    }
+
+    const scriptPath = path.join(codemodsPath, `${codemod}.js`)
+    if (!fs.pathExistsSync(scriptPath)) {
+        log.error(`The package "${package}" does not have a codemod called "${codemod}"`)
+    }
+
+    log.info('Starting running the codemod\n')
+
+    const result = spawn.sync('npx', [
+        '--no-install',
+        'jscodeshift',
```

Comment:
It's probably unlikely that `jscodeshift` will be installed locally, right?

Reply:
Could we make it a dependency of this tool?

Reply:
It's in the `dependencies`. Do you mean I should move it to `peerDependencies`?

Reply:
Right, I meant call it programmatically instead of using `npx`.  Unfortunately I don't think `npx --no-install` will work here... if you run `npx @dhis2/code-mods xyz` or `d2 utils codemods xyz` in some directly, spawning a new `npx no-install` in that directory will fail because it's in a totally separate context (I think).  I imagine this package won't itself be a dependency of apps, since it's used only for one-off operations.  Maybe I'm wrong about how `npx` will behave there, though - @varl has used it with `cli-style` but that's often / usually installed locally.

Reply:
Good point. I've switched out the call to `spawn.sync` with calling jscodeshift's runner programmatically